plugins/collections/create.js

```diff
@@ -30,12 +30,26 @@ module.exports = () => ({
                         throw boom.notFound(`User ${username} does not exist`);
                     }
 
-                    return collectionFactory.create({
-                        name: request.payload.name,
-                        description: request.payload.description,
-                        userId: user.id,
-                        pipelineIds: request.payload.pipelineIds
-                    });
+                    const config = Object.assign({}, request.payload);
+
+                    config.userId = user.id;
+                    // Check that the pipelines exist for the pipelineIds specified.
+                    if (request.payload.pipelineIds) {
+                        const { pipelineFactory } = request.server.app;
+
+                        return Promise.all(request.payload.pipelineIds.map(pipelineId =>
```

Comment:
Curious. Could this also be done with `request.payload.pipelineIds.map(pipelineFactory.get)`?